lib/net.js

```diff
@@ -864,6 +870,9 @@ function afterWrite(status, handle, req, err) {
   if (self !== process.stderr && self !== process.stdout)
     debug('afterWrite', status);
 
+  if (req.async)
+    self[kLastWriteQueueSize] = 0;
```

Comment:
Just to make sure my understanding here is correct:

This line is correct because only one write is active at a time, so once a write is carried out successfully, the write queue size must be zero.

It is only necessary for async writes, because after sync writes the write queue size is 0 anyway.

Right?

Reply:
Correct. If there's ever more than one write, this is all terribly broken anyway and we have bigger problems than tracking the queue size.